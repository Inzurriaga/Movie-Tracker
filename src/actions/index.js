// what gets sent
export const getMovies = (movies) => ({
  type: 'GET_MOVIES',
  movies
});

export const addUser = (userInfo) => ({
  type: "UPDATE_USER",
  userInfo
})

export const getGenres = (genre) => ({
  type: "GET_GENRES",
  genre
})
