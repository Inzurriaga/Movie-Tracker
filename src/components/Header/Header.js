import React, { Component} from 'react';
import SignIn from "../../containers/SignIn/SignIn";

class Header extends Component {
  constructor() {
    super();
    this.state = {
      show: false
    }
  }

  showModal = () => {
    this.setState({ show: true });
  };

  hideModal = () => {
    this.setState({ show: false });
  };

  render() {
    return (
      <div className="Header">
        <section className="Logo">
          <span>Logo</span>
        </section>
        <section className="Login-Section">
          <SignIn show={this.state.show} handleClose={this.hideModal}>

          <button type="button" onClick={this.showModal}>
            open
          </button>
          
          <SignIn />
        </section>

      </div>
    )
  }
}

export default Header;
