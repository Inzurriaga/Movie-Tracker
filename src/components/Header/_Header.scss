.Header {
  background:#0c0f29;
  height: 20px;
  width: 100%
}
