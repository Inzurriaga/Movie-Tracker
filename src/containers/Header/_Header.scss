.Header {
  background:#0c0f29;
  box-sizing: border-box;
  padding: 1em 2em;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;

  .Header-Nav {
    color:#fff;
    display: flex;
    div {
      padding-right: 1em;
      h2 {
        margin: 0;
        // font-family: 'Barlow Condensed', sans-serif;
        font-family: $font-Abel;
        // font-family: 'PT Sans Narrow', sans-serif;
        // font-family: 'Open Sans Condensed', sans-serif;
        // font-family: $font-Abel;
      }
    }
  }

  .LoginBtn {
    display: flex;
    align-items: center;

    h4 {
      font-family: $font-Abel;
      margin: 0;
      color: #fff;
      padding-right: 1em;
    }

    button {
      background: none;
      border-radius: 50%;
      padding: 5px 7px;
      border: solid 2px #fff;
      opacity: 1;
      cursor: pointer;
      outline: none;
      transition: all .5s ease;

      &:hover {
        opacity: .7;
      }
    }

    i {
      color:#fff;
    }
  }


  .Modal-Container {

    &.modal-show {
      position: fixed;
      top: 0;
      left: 0;
      width:100%;
      height: 100%;
      background: rgba(0, 0, 0, 0.6);
      z-index: 999;
    }

    &.modal-hide {
      display: none;
    }
  }

}
