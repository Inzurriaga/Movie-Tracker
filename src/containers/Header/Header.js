import React, { Component} from 'react';
import SignIn from "../SignIn/SignIn";
import { Link, NavLink } from 'react-router-dom'
import { connect } from 'react-redux'
import { signOutUser, signOutFavorites } from '../../actions';

export class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      show: false
    }
  }

  showModal = () => {
    this.setState({ show: true });
  };

  hideModal = () => {
    this.setState({ show: false });
  };

  signOut = () => {
    this.props.signOutUser();
    this.props.signOutFavorites();
  }

  // need utilities file for multiple methods
  isEmpty = (obj) => {
    for(var key in obj) {
        if(obj.hasOwnProperty(key))
            return true;
    }
    return false;
  }

  render() {
    console.log(this.props.user);
    const showHideClassName = this.state.show ? "Modal-Container modal-show" : "Modal-Container modal-hide";
    return (
      <div className="Header">
        <nav className="Header-Nav">
          <Link key="Home" to="/"><h2>Home</h2></Link>
          <Link key="All_Movies" to="/movies/allMovies"><h2>All Genres</h2></Link>
          {
            this.isEmpty(this.props.user) && <Link key="Favorites" to="/favorites"><h2>Favorites</h2></Link>
          }
        </nav>
        <section className="LoginBtn">
        { !this.isEmpty(this.props.user) &&
          <div className="User-Info-Box">
            <h4 className="User-Info">Sign In</h4>
            <button type="button" onClick={this.showModal}>
              <i className="fas fa-user"></i>
            </button>
          </div>
        }
        { this.isEmpty(this.props.user) &&
          <div className="User-Info-Box">
            <h4 className="User-Info">Welcome {this.props.user.name}</h4>
            <NavLink to="/" type="button" onClick={this.signOut}>
              <i className="fas fa-sign-out-alt"></i>
            </NavLink>
          </div>
        }
        </section>
        <section className={showHideClassName}>
          <SignIn className="SignIn" show={this.state.show} hideModal={this.hideModal} />
        </section>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  favorites: state.favorites,
})

export const mapDispatchToProps = (dispatch) => ({
  signOutUser: (signOut) => dispatch(signOutUser(signOut)),
  signOutFavorites: (favorites) => dispatch(signOutFavorites(favorites))
})


export default connect(mapStateToProps, mapDispatchToProps)(Header);
