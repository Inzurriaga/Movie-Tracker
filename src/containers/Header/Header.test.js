import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import { signOutUser, signOutFavorites } from '../../actions/index';

let mockPropsUser = {};
let mockSignOutUser = jest.fn()
let mockSignOutFavorites = jest.fn()

describe('Header', () => {

  describe('Header Component', () => {
    let headerComponenet;

    beforeEach(() => {
      headerComponenet = shallow(<Header user={mockPropsUser}
                                         signOutUser={mockSignOutUser}
                                         signOutFavorites={mockSignOutFavorites}
                                  />)
    })

    it('should match the snapshot', () => {
      expect(headerComponenet).toMatchSnapshot()
    })

    it('should have default properties', () => {
      expect(headerComponenet.state()).toEqual({
        show: false
      })
    })

    it('should change its show state when showModal is clicked', () => {
      headerComponenet.setState({
        show: false
      })
      expect(headerComponenet.state("show")).toEqual(false);
      headerComponenet.instance().showModal();
      expect(headerComponenet.state("show")).toEqual(true);
      expect(headerComponenet).toMatchSnapshot()
    })

    it('should change its show state when hideModal is clicked', () => {
      headerComponenet.setState({
        show: true
      })
      expect(headerComponenet.state("show")).toEqual(true);
      headerComponenet.instance().hideModal();
      expect(headerComponenet.state("show")).toEqual(false);
      expect(headerComponenet).toMatchSnapshot()
    })

    it('should change render and show a welcome user when a user has logged in', () => {
      // expect(mockPropsUser).toEqual({})
      headerComponenet.setProps({ user: {id: 7, name: "a", password: "a", email: "a"}})
      expect(headerComponenet).toMatchSnapshot()
    })

    it('should change render and show a welcome user when a user has logged in', () => {
      headerComponenet.instance().signOut()

      expect(mockSignOutUser).toHaveBeenCalled()
      expect(mockSignOutFavorites).toHaveBeenCalled()
    })


  })

  describe('Header mapStateToProps', () => {

    it('should return an object of default states', () => {
      const mockState = {
        user: {id: 7, name: "a", password: "a", email: "a"},
        favorites: [{page: 1, total_pages: 1176, total_results: 23512}],
        filter: []
      }
      const expected = {
        user: {id: 7, name: "a", password: "a", email: "a"},
        favorites: [{page: 1, total_pages: 1176, total_results: 23512}],
      }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected)
    })

  })

  describe('Header mapDispatchToProps', () => {

    it('should call dispatch for signOutUser using a function from mapDispatchToProps', () => {
      const mockDispatchUser = jest.fn();
      const mockUser = {id: 7, name: "a", password: "a", email: "a"};

      const actionToDispatch = signOutUser(mockUser)
      const mappedProps = mapDispatchToProps(mockDispatchUser);
      mappedProps.signOutUser(mockUser);
      expect(mockDispatchUser).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch for signOutFavorites using a function from mapDispatchToProps', () => {
      const mockDispatchFavorites = jest.fn();
      const mockFavorites = [{page: 1, total_pages: 1176, total_results: 23512}];

      const actionToDispatch = signOutFavorites(mockFavorites)
      const mappedProps = mapDispatchToProps(mockDispatchFavorites);
      mappedProps.signOutFavorites(mockFavorites);
      expect(mockDispatchFavorites).toHaveBeenCalledWith(actionToDispatch)
    })
  })

})
