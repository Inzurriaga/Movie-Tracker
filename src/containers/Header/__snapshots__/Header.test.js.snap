// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Component should change its show state when hideModal is clicked 1`] = `
<div
  className="Header"
>
  <nav
    className="Header-Nav"
  >
    <Link
      key="Home"
      to="/"
    >
      <h2>
        Home
      </h2>
    </Link>
    <Link
      key="All_Movies"
      to="/movies/allMovies"
    >
      <h2>
        All Genres
      </h2>
    </Link>
    <Link
      key="Favorites"
      to="/favorites"
    >
      <h2>
        Favorites
      </h2>
    </Link>
  </nav>
  <section
    className="LoginBtn"
  >
    <div
      className="User-Info-Box"
    >
      <h4
        className="User-Info"
      >
        Sign In
      </h4>
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-user"
        />
      </button>
    </div>
  </section>
  <section
    className="Modal-Container modal-hide"
  >
    <Connect(SignIn)
      className="SignIn"
      hideModal={[Function]}
      show={false}
    />
  </section>
</div>
`;

exports[`Header Header Component should change its show state when showModal is clicked 1`] = `
<div
  className="Header"
>
  <nav
    className="Header-Nav"
  >
    <Link
      key="Home"
      to="/"
    >
      <h2>
        Home
      </h2>
    </Link>
    <Link
      key="All_Movies"
      to="/movies/allMovies"
    >
      <h2>
        All Genres
      </h2>
    </Link>
    <Link
      key="Favorites"
      to="/favorites"
    >
      <h2>
        Favorites
      </h2>
    </Link>
  </nav>
  <section
    className="LoginBtn"
  >
    <div
      className="User-Info-Box"
    >
      <h4
        className="User-Info"
      >
        Sign In
      </h4>
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-user"
        />
      </button>
    </div>
  </section>
  <section
    className="Modal-Container modal-show"
  >
    <Connect(SignIn)
      className="SignIn"
      hideModal={[Function]}
      show={true}
    />
  </section>
</div>
`;

exports[`Header Header Component should change render and show a welcome user when a user has logged in 1`] = `
<div
  className="Header"
>
  <nav
    className="Header-Nav"
  >
    <Link
      key="Home"
      to="/"
    >
      <h2>
        Home
      </h2>
    </Link>
    <Link
      key="All_Movies"
      to="/movies/allMovies"
    >
      <h2>
        All Genres
      </h2>
    </Link>
    <Link
      key="Favorites"
      to="/favorites"
    >
      <h2>
        Favorites
      </h2>
    </Link>
  </nav>
  <section
    className="LoginBtn"
  >
    <div
      className="User-Info-Box"
    >
      <h4
        className="User-Info"
      >
        Welcome 
        a
      </h4>
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-sign-out-alt"
        />
      </button>
    </div>
  </section>
  <section
    className="Modal-Container modal-hide"
  >
    <Connect(SignIn)
      className="SignIn"
      hideModal={[Function]}
      show={false}
    />
  </section>
</div>
`;

exports[`Header Header Component should match the snapshot 1`] = `
<div
  className="Header"
>
  <nav
    className="Header-Nav"
  >
    <Link
      key="Home"
      to="/"
    >
      <h2>
        Home
      </h2>
    </Link>
    <Link
      key="All_Movies"
      to="/movies/allMovies"
    >
      <h2>
        All Genres
      </h2>
    </Link>
    <Link
      key="Favorites"
      to="/favorites"
    >
      <h2>
        Favorites
      </h2>
    </Link>
  </nav>
  <section
    className="LoginBtn"
  >
    <div
      className="User-Info-Box"
    >
      <h4
        className="User-Info"
      >
        Sign In
      </h4>
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-user"
        />
      </button>
    </div>
  </section>
  <section
    className="Modal-Container modal-hide"
  >
    <Connect(SignIn)
      className="SignIn"
      hideModal={[Function]}
      show={false}
    />
  </section>
</div>
`;
